doc/source/management/data_at_rest_encryption.rst

```diff
@@ -11,244 +11,422 @@ This feature is considered **BETA** quality.
+continue operating with same MASTER-key.
+
+Compatibility
+*************
+
+Keyring (or, more generally, the |PXC| SST process) is backward compatible, as
+in higher version JOINER can join from lower version DONOR, but not vice-versa.
+More details are covered below, in `Upgrade and compatibility issues`_ section.
+
+.. note:: Since |PXC| 5.6 didn't have encrypted tablespace, no major
+   upgrade scenario for data-at-rest encryption is possible from it.
+
+Configuring PXC to use keyring_vault plugin
+===========================================
+
+keyring_vault
+-------------
 
-A new option, implemented since |Percona Server| :rn:`5.7.20-19`, is
-encryption of binary and relay logs, triggered by the
-:variable:`encrypt_binlog` variable.
+The ``keyring_vault`` plugin is supported starting from PXC 5.7.22. This plugin
+allows storing the master-key in vault-server (vs. local file as in case of
+``keyring_file``). 
 
-Besides turning :variable:`encrypt_binlog` ``ON``, this feature requires both
-`master_verify_checksum
-<https://dev.mysql.com/doc/refman/5.7/en/replication-options-binary-log.html#sysvar_master_verify_checksum>`_
-and `binlog_checksum
-<https://dev.mysql.com/doc/refman/5.7/en/replication-options-binary-log.html#sysvar_binlog_checksum>`_
-variables to be turned ``ON``.
+.. warning:: rsync doesn't support ``keyring_vault``, and SST on JOINER is
+   aborted if rsync is used on the node with ``keyring_vault`` configured. 
 
-While replicating, master sends the stream of decrypted binary log events to a
-slave (SSL connections can be set up to encrypt them in transport). That said,
-masters and slaves use separate keyring storages and are free to use differing
-keyring plugins.
+Configuration
+*************
 
-Dumping of encrypted binary logs involves decryption, and can be done using
-``mysqlbinlog`` with ``--read-from-remote-server`` option.
+Configuration options are same as `upstream <https://www.percona.com/doc/percona-server/5.7/management/data_at_rest_encryption.html#keyring-vault-plugin>`_. The ``my.cnf`` configuration file should contain following options:
 
-.. note:: Taking into account that ``--read-from-remote-server`` option  is only
-   relevant to binary logs, encrypted relay logs can not be dumped/decrypted
-   in this way.
+.. code-block:: text
 
-System Variables
-----------------
+   [mysqld]
+   early-plugin-load="keyring_vault=keyring_vault.so"
+   keyring_vault_config="<PATH>/keyring_vault_n1.conf"
 
-.. variable:: encrypt_binlog
+Also ``keyring_vault_n1.conf`` file contents should be :
 
-  :version 5.7.20-19: Implemented
-  :cli: ``--encrypt-binlog``
-  :dyn: No
-  :scope: Global
-  :vartype: Boolean
-  :default: ``OFF``
+.. code-block:: text
 
-The variable turns on binary and relay logs encryption.
+   vault_url = http://127.0.0.1:8200
+   secret_mount_point = secret1
+   token = e0345eb4-35dd-3ddd-3b1e-e42bb9f2525d
+   vault_ca = /data/keyring_vault_confs/vault_ca.crt
 
```

Comment:
Add link to upstream so that user know how to use these options. Upstream doc has detailed description of these options.

Reply:
done